### PR TITLE
Add supply war info to METER_SUPPLY_VALUE_DESC

### PR DESCRIPTION
**Pull request by agrrr3**
supply.cpp#460 there is some tie breaking stuff (bonus from colonies, outpost, multiple sources of supply...)
...
we will document the tie breaker conditions somewhere in pedia

### CODE REVIEW

**Comment by agrrr3** [comment]
@geoffthemedio would you review, please

File: default/stringtables/en.txt [hunk 1]
```diff
@@ -6433,7 +6433,16 @@ A set of core supply starlanes for an empire (the set of least-jump starlanes co
 
 Supply lines can also be used to refill the [[metertype METER_FUEL]] supply of ships.
 
-Supply lines will be obstructed by armed enemy ships belonging to empires with which an empire is at [[encyclopedia WAR_TITLE]]. At the opposite end of the spectrum, empires in an [[encyclopedia ALLIANCE_TITLE]] will share supply lines and all their associated benefits.'''
+Supply lines will be obstructed by armed enemy ships belonging to empires with which an empire is at [[encyclopedia WAR_TITLE]]. At the opposite end of the spectrum, empires in an [[encyclopedia ALLIANCE_TITLE]] will share supply lines and all their associated benefits.
```
Comment (geoffthemedio) [inline default/stringtables/en.txt]:
```suggestion
Empires in an [[encyclopedia ALLIANCE_TITLE]] use eachother's supply lines.

The extension of supply lines from source planets into a system can be obstructed by armed enemy ships in that system and that are in fleets set to Obstruct Passage or Initiate Combat. Enemy ships are those owned by empires that are at [[encyclopedia WAR_TITLE]] with the supply line's empire.
```

File: default/stringtables/en.txt [hunk 2]
```diff
@@ -6433,7 +6433,16 @@ A set of core supply starlanes for an empire (the set of least-jump starlanes co
...
+
+If multiple empires compete for supply in a system, the empire with the highested propagated supply will win. In order to break ties, a bonus per empire is calculated and added to the propagated supply before comparison.
```
Comment (geoffthemedio) [inline default/stringtables/en.txt]:
```suggestion
If multiple empires compete for supply in a system (without any obstructions to either's supply propagation), the empire with the highested propagated supply range will supply that system. To break ties, a bonus per empire is calculated and added to the propagated supply before comparison.
```

File: default/stringtables/en.txt [hunk 3]
```diff
@@ -6433,7 +6433,16 @@ A set of core supply starlanes for an empire (the set of least-jump starlanes co
...
+• Has a colony in the system: +0.5
+• Has an outpost in the system: +0.3
```
Comment (geoffthemedio) [inline default/stringtables/en.txt]:
```suggestion
• Has an outpost (but no colony) in the system: +0.3
```

File: default/stringtables/en.txt [hunk 4]
```diff
@@ -6433,7 +6433,16 @@ A set of core supply starlanes for an empire (the set of least-jump starlanes co
...
+* Has supply sources in the system: +(0.001 * SUM(own supply in the system))
+* Has supply sources in the system: +(0.00001 * SUM(own max supply in the system))
+* Total supply: +(0.00000001 * SUM(own supply in the empire))
+* Has supplying source outside the system: +(0.0001 * MAX(1.0, MAX(distance to supplying system)))
```
Comment (geoffthemedio) [inline default/stringtables/en.txt:6444]: * Distance to the source of supply that, after propagation, is the strongest in the system. That is, a 4 strength source 4 jumps away is stronger than a 3 strength srouce 3 jumps away: +(0.0001 * distance to supplying system)

**Comment by geoffthemedio** [comment]
not sure that giving specifics of the calculation (eg. * 0.00001 or similar) is necessary...

File: default/stringtables/en.txt (hunk 4 above)
Comment (geoffthemedio) [inline default/stringtables/en.txt]:
```suggestion
* Distance to the source of supply that reaches the system and that, before propagation, is the strongest. That is, a 4 strength source 4 jumps away is stronger than a 3 strength srouce 3 jumps away: +(0.0001 * distance to supplying system)
```

**Comment by agrrr3** [comment]
> not sure that giving specifics of the calculation (eg. * 0.00001 or similar) is necessary...

at least some indication of order of magnitude is necessary, as the last three can reasonably assumed to interact.
if this is intended as a strict hierarchy, we wont need the specifics, but probably we need to change the bonus calculation

File: default/stringtables/en.txt (hunk 4 above)
Comment (agrrr3) [reply default/stringtables/en.txt]: i had a hard time analysing what empire_supply_it actually contains.

Is `empire_supply_it->second.second` the original (not propagated) supply strength? so a 4 strength system 2 jumps away is stronger than a 3 strength system 3 jumps away? in that case the distance does not matter.
 
Or does it really contain the distance to the strongest supply source? 

anyway, i was not sure why there is a maximum calculation: `max(1.0f, empire_supply_it->second.second);` 
there would be only a problem if the value was negative; if it is zero the following division result should also be zero.(!?)

Comment (agrrr3) [reply default/stringtables/en.txt]: ah, just saw your explanations on the forum for the dominant supply source, thanks.

**Comment by geoffthemedio** [comment]
> if this is intended as a strict hierarchy

not really. just stuff to break the ties in some sensible way.

File: default/stringtables/en.txt [hunk 5]
```diff
@@ -6433,7 +6433,18 @@ A set of core supply starlanes for an empire (the set of least-jump starlanes co
 
 Supply lines can also be used to refill the [[metertype METER_FUEL]] supply of ships.
 
-Supply lines will be obstructed by armed enemy ships belonging to empires with which an empire is at [[encyclopedia WAR_TITLE]]. At the opposite end of the spectrum, empires in an [[encyclopedia ALLIANCE_TITLE]] will share supply lines and all their associated benefits.'''
+Empires in an [[encyclopedia ALLIANCE_TITLE]] use each other's supply lines.
+The extension of supply lines from source planets into a system can be obstructed by armed enemy ships in that system and that are in fleets set to [[FLEET_OBSTRUCTIVE]] or [[FLEET_AGGRESSIVE]]. Enemy ships are those owned by empires that are at [[encyclopedia WAR_TITLE]] with the supply line's empire.
```
Comment (geoffthemedio) [inline default/stringtables/en.txt]:
```suggestion
The extension of supply lines from source planets into a system can be obstructed by armed enemy ships that are in that system and that are in fleets set to [[FLEET_OBSTRUCTIVE]] or [[FLEET_AGGRESSIVE]]. Enemy ships are those owned by empires that are at [[encyclopedia WAR_TITLE]] with the supply line's empire and neutral / monster ships.
```

From testing and the code, I think monsters also qualify for obstruction of supply propagation.

**Pull request merged**